kairon/events/definitions/analytic_pipeline_handler.py

```
@@ -0,0 +1,222 @@
+            logger.error(e)
+            if event_id:
+                AnalyticsPipelineProcessor.delete_task(event_id, self.bot)
+            raise AppException(e)
+
+
+    def delete_schedule(self, event_id: Text):
+        """Delete a scheduled analytics pipeline event."""
+        try:
+            if not Utility.is_exist(AnalyticsPipelineConfig, raise_error=False, bot=self.bot,
+                                    id=event_id, status=True):
+                raise AppException("Notification settings not found!")
```

Ruff (0.14.7)

Abstract `raise` to an inner function

(TRY301)